src/typography/helper.ts

```diff
@@ -0,0 +1,98 @@
+import { css } from "styled-components";
+import { TypographyProps, TypographySizeType, TypographyWeight } from "./types";
+import { TypographyStyle } from "./typography-style";
+import { Colour, Font } from "../theme";
+import { StyledComponentProps } from "../theme/helpers";
+
+const getResolvedTypographyWeight = (
+    weight: TypographyWeight,
+    props: StyledComponentProps
+): string => {
+    const weightMap: Record<number, string> = {
+        300: "light",
+        400: "regular",
+        600: "semibold",
+        700: "bold",
+    };
+
+    // If resolvedWeight is a string that is a number for eg "400", convert it to a number
+    const numericWeight =
+        typeof weight === "string" && !isNaN(Number(weight))
+            ? Number(weight)
+            : weight;
+
+    // Map it to its string equivalent
+    const mappedWeight = weightMap[numericWeight as number] || numericWeight;
+    const finalWeight = Font[`weight-${mappedWeight}`];
+
+    // If final weight is a function, resolve it with props
+    return typeof finalWeight === "function" ? finalWeight(props) : finalWeight;
+};
+
+export const getTypographyStyle = (
+    type: TypographySizeType,
+    weight: TypographyWeight,
+    paragraph = false
+) => {
+    return (props: any) => {
+        const attrs = TypographyStyle[type];
+
+        const resolvedWeight = getResolvedTypographyWeight(weight, props);
+
+        // Check if function if so resolve with props
+        const fontSize =
+            typeof attrs.fontSize === "function"
+                ? attrs.fontSize(props)
+                : attrs.fontSize;
+
+        // Make it a int for calc
+        const fontSizeValue = parseFloat(fontSize);
+        const fontSizeUnit = fontSize.replace(fontSizeValue.toString(), "");
+
+        // Add extra margin for paragraphs
+        const getMarginBottomStyle = () => {
+            const marginBottomScale = paragraph ? 1.05 : 0;
+            return css`
+                margin-bottom: ${fontSizeValue * marginBottomScale}${fontSizeUnit};
+            `;
+        };
+
+        return css`
+            font-size: ${fontSize};
+            line-height: ${attrs.lineHeight};
+            letter-spacing: ${attrs.letterSpacing || 0};
+            font-weight: ${resolvedWeight};
```

Comment:
instead of individually picking out the styles, we can reuse the `Typography` tokens instead?

for example, HeaderXL will use `header-xl-<weight>` where weight comes from the prop

Reply:
Just to check if i remove this if i am not wrong i would have to create a helper function or within the code itself do like a iteration to convert the header-xl etc values to css properties. Is that intended or is there another approach ?

Reply:
you can simplify the current implementation of the Typography namespace, I am envisioning something like

```
export const HeaderXXL = createHeader("h1", "header-xxl");

// given HeaderXL + `weight="bold"`, it results in this:
styled(tag)`
  ${Typography["header-xl-bold"]}
`
```